contracts/ligo/src/fixed_price_sale_market_tez.mligo

```diff
@@ -58,29 +65,40 @@ let transfer_tez (price, seller : tez * address) : operation =
         Tezos.transaction () Tezos.amount seller_account
 
 let buy_token(sale, storage: sale_param_tez * storage) : (operation list * storage) =
-  let sale_price = match Big_map.find_opt sale storage with
+  let sale_price = match Big_map.find_opt sale storage.sales with
   | None -> (failwith "NO_SALE": tez)
   | Some s -> s
   in let tx_ops = transfer_tez(sale_price, sale.sale_seller) in
   let tx_nft_op = transfer_nft(sale.sale_token.token_for_sale_address, sale.sale_token.token_for_sale_token_id, Tezos.self_address, Tezos.sender) in
-  let new_s = Big_map.remove sale storage in
+  let new_s = { storage with sales = Big_map.remove sale storage.sales } in
   (tx_ops :: tx_nft_op :: []), new_s
 
 let deposit_for_sale(sale_token, price, storage: sale_token_param_tez * tez * storage) : (operation list * storage) =
     let transfer_op =
       transfer_nft (sale_token.token_for_sale_address, sale_token.token_for_sale_token_id, Tezos.sender, Tezos.self_address) in
     let sale_param = { sale_seller = Tezos.sender; sale_token = sale_token } in
-    let new_s = Big_map.add sale_param price storage in
+    let new_s = { storage with sales = Big_map.add sale_param price storage.sales } in
     (transfer_op :: []), new_s
 
-let cancel_sale(sale, storage: sale_param_tez * storage) : (operation list * storage) = match Big_map.find_opt sale storage with
+let cancel_sale(sale, storage: sale_param_tez * storage) : (operation list * storage) = match Big_map.find_opt sale storage.sales with
     | None -> (failwith "NO_SALE" : (operation list * storage))
     | Some price -> if sale.sale_seller = Tezos.sender then
```

Comment:
To cancel you must both be an admin and the seller? Shouldn't the logic be, an admin OR the seller can cancel?

Reply:
Ah, good catch. Thanks. I'll change the logic accordingly.

Reply:
Shouldn't only admin be able to make a sale as well? Then the logic only needs to be, only a seller (who also is necessarily the admin) can cancel.